async-nats/src/lib.rs

```diff
@@ -821,6 +817,11 @@ pub async fn connect_with_options<A: ToServerAddrs>(
         .await
         .map_err(|_| io::Error::new(io::ErrorKind::Other, "failed to send ping"))?;
 
+    task::spawn(async move {
```

Comment:
Moving this can end up blocking? if the channel is configured to be of size 1

Reply:
yeah, should not be here. Was moved to read error from connect, but that i removed too.